src/java.base/share/classes/java/lang/AtomicAccess.java

```diff
@@ -26,38 +26,52 @@
  * when loading or storing any value of the class to a field or array
- * element.  Normally, only fields declared {@code volatile} are
- * protected against structure tearing, but a class that implements
- * this marker interface will never have its values torn, even when
+ * element.  Normally, only naturally atomic fields and fields declared
+ * {@code volatile} are always atomic, but a class that implements
+ * this marker interface will always be accessed atomically, even when
  * they are stored in array elements or in non-{@code volatile}
  * fields, and even when multiple threads perform racing writes.
  *
- * <p> An primitive instance of multiple components is said to be "torn"
- * when two racing threads compete to write those components, and one
- * thread writes some components while another thread writes other
- * components, so a subsequent observer will read a hybrid composed,
- * as if "out of thin air", of field values from both racing writes.
- * Tearing can also occur when the effects of two non-racing writes
- * are observed by a racing read.  In general, structure tearing
- * requires a read and two writes (initialization counting as a write)
- * of a multi-component value, with a race between any two of the
- * accesses.  The effect can also be described as if the Java memory
- * model break up primitive classinstance reads and writes into reads and
+ * <p> A primitive instance of multiple components can experience both
+ * transient and persistent access atomicity failures.
+ *
+ * <p> Transient failures usually arise from write-read conflicts:
+ * when one thread writes the components one-by-one, and another
+ * thread reads the components one-by-one. In doing so, reader
+ * observes the intermediate state of the primitive instance.
+ * This failure is transient, since "after" (in memory model sense)
+ * the writer finishes its writes, the instance is observed in full
+ * by any subsequent observer.
+ *
+ * <p> Permanent failures usually arise from write-write conflicts:
+ * when two threads compete to write the components, and one thread
+ * writes some components while another thread writes other
+ * components. This failure is permanent: as every subsequent observer
+ * will read a hybrid composed of field values from both racing writes.
+ *
+ * Both these effects can be described as if the Java memory model
+ * break up primitive class instance reads and writes into reads and
  * writes of their various fields, as it does with longs and doubles
  * (JLS 17.7).
  *
- * <p> In extreme cases, the hybrid observed after structure tearing
+ * <p> In extreme cases, the hybrid observed under non-atomic access
  * might be a value which is impossible to construct by normal means.
  * If data integrity or security depends on proper construction,
- * the class should be declared as implementing {@code NonTearable}.
+ * the class should be declared as implementing {@code AtomicAccess}.
+ *
```

Comment:
This comment is useful, because it distinguishes non-tearable from the
de-facto standard notion (in the JDK) of "atomic", which pertains to
"compound operations".  Atomic compound operations relate multiple
states of a variable in a consistent sequence sequence.  This is a
strictly stronger notion than access atomicity or non-tearability.

I support adding such a comment here, but also note that its
presence in the PR is evidence that the type name should
*not* be changed to `AtomicAccess`, precisely because
users are likely to confuse *this new* use of atomic with
*existing old* uses, all of which imply the stronger concept.